src/fork_gc.c

```diff
@@ -0,0 +1,757 @@
+#include "fork_gc.h"
+#include "util/arr.h"
+#include "search_ctx.h"
+#include "inverted_index.h"
+#include "redis_index.h"
+#include "numeric_index.h"
+#include "tag_index.h"
+#include "tests/time_sample.h"
+#include <stdlib.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <wait.h>
+
+static void ForkGc_updateStats(RedisSearchCtx *sctx, ForkGCCtx *gc, size_t recordsRemoved,
+                               size_t bytesCollected) {
+  sctx->spec->stats.numRecords -= recordsRemoved;
+  sctx->spec->stats.invertedSize -= bytesCollected;
+  gc->stats.totalCollected += bytesCollected;
+}
+
+static void ForkGc_FDWriteLongLong(int fd, long long val) {
```

Comment:
Check the return value

Reply:
there is an assert about the number of bytes written ..